### ISSUE TITLE
Screen transitioning color causing next screen to fade out the same color...

### ISSUE THREAD

**Issue by born2snipe** [mini2Dx/mini2Dx#77]
So the weirdness is if you are using the same `Color` instance in the `FadeInTransition` on the screen instance you faded into those items being drawn with that color will fade out. Below is a minimal example to show exactly the behavior I am seeing. 

```java

import com.badlogic.gdx.graphics.Color;
import org.mini2Dx.core.game.GameContainer;
import org.mini2Dx.core.game.ScreenBasedGame;
import org.mini2Dx.core.graphics.Graphics;
import org.mini2Dx.core.screen.BasicGameScreen;
import org.mini2Dx.core.screen.GameScreen;
import org.mini2Dx.core.screen.ScreenManager;
import org.mini2Dx.core.screen.Transition;
import org.mini2Dx.core.screen.transition.FadeInTransition;
import org.mini2Dx.core.screen.transition.FadeOutTransition;

import static com.badlogic.gdx.graphics.Color.BLACK;

public class BugGame extends ScreenBasedGame {

    @Override
    public void initialise() {
        addScreen(new FirstScreen());
        addScreen(new SecondScreen());
    }

    @Override
    public int getInitialScreenId() {
        return FirstScreen.ID;
    }

    private class FirstScreen extends BasicGameScreen {
        private static final int ID = 0;

        @Override
        public void initialise(GameContainer gc) {

        }

        @Override
        public void update(GameContainer gc, ScreenManager<? extends GameScreen> screenManager, float delta) {
            screenManager.enterGameScreen(SecondScreen.ID, new FadeOutTransition(), new FadeInTransition(BLACK));
        }

        @Override
        public void interpolate(GameContainer gc, float alpha) {

        }

        @Override
        public void render(GameContainer gc, Graphics g) {
        }

        @Override
        public int getId() {
            return ID;
        }
    }


    private class SecondScreen extends BasicGameScreen {
        private static final int ID = 1;
        private float waitTime = 0;
        private float maxWaitTime = 5f;

        @Override
        public void initialise(GameContainer gc) {
        }

        @Override
        public void preTransitionIn(Transition transitionIn) {
            waitTime = 0;
        }

        @Override
        public void update(GameContainer gc, ScreenManager<? extends GameScreen> screenManager, float delta) {
            waitTime += delta;

            if (waitTime >= maxWaitTime) {
                screenManager.enterGameScreen(FirstScreen.ID, new FadeOutTransition(), new FadeInTransition());
            }
        }

        @Override
        public void interpolate(GameContainer gc, float alpha) {

        }

        @Override
        public void render(GameContainer gc, Graphics g) {
            g.setBackgroundColor(Color.WHITE);

            // this one fades away
            g.setColor(BLACK);
            g.fillRect(10f, 10f, 100f, 100f);

            g.setColor(Color.RED);
            g.fillRect(120f, 10f, 100f, 100f);

            // this one stays on the screen
            g.setColor(new Color(0, 0, 0, 1));
            g.fillRect(10f, 120f, 100f, 100f);
        }

        @Override
        public int getId() {
            return ID;
        }
    }
}
```

**Comment by tomcashman** [comment]
Thanks for reporting :)

Currently the fade transitions modify the color instance so as a workaround you'll need to [copy the Color](https://libgdx.badlogicgames.com/ci/nightlies/docs/api/com/badlogic/gdx/graphics/Color.html#cpy--) and pass the new instance to the transition.

I'll add a fix next week when I'm back from EGX :)

**Comment by tomcashman** [comment]
This is fixed in 1.7.14 :)
